src/sagemaker/workflow/model_step.py

```diff
@@ -0,0 +1,216 @@
+            depends_on (List[str] or List[Step]): A list of `Step` names or `Step` instances
+                that this `ModelStep` depends on.
+                If a listed `Step` name does not exist, an error is returned (default: None).
+            retry_policies (List[RetryPolicy] or Dict[str, List[RetryPolicy]]): The list of retry
+                policies for the `ModelStep`.
+            display_name (str): The display name of the `ModelStep`.
+                The display name provides better UI readability. (default: None).
+            description (str): The description of the `ModelStep` (default: None).
+        """
+        if not isinstance(model.sagemaker_session, PipelineSession):
+            raise ValueError(
+                "To correctly configure a ModelStep, "
+                "the sagemaker_session of the model must be a PipelineSession."
+            )
+        self.name = name
+        self.step_args = step_args
+        self.model = model
+        self.depends_on = depends_on
+        self.retry_policies = retry_policies
+        self.display_name = display_name
+        self.description = description
+        self.steps: List[Step] = []
+        self._create_model_args = self.step_args.get(CREATE_MODEL_REQUEST, None)
+        self._regis_model_args = self.step_args.get(CREATE_MODEL_PACKAGE_REQUEST, None)
+        self._create_model_runtime_repack = (
+            self._create_model_args.pop(NEED_RUNTIME_REPACK, None)
+            if self._create_model_args
+            else None
+        )
+        self._regis_model_runtime_repack = (
+            self._regis_model_args.pop(NEED_RUNTIME_REPACK, None)
+            if self._regis_model_args
+            else None
+        )
+
+        if isinstance(retry_policies, dict):
+            self._create_model_retry_policies = retry_policies.get(
+                _CREATE_MODEL_RETRY_POLICIES, None
+            )
+            self._register_model_retry_policies = retry_policies.get(
+                _REGIS_MODEL_RETRY_POLICIES, None
+            )
+            self._repack_model_retry_policies = retry_policies.get(
+                _REPACK_MODEL_RETRY_POLICIES, None
+            )
+        else:
+            self._create_model_retry_policies = retry_policies
+            self._register_model_retry_policies = retry_policies
+            self._repack_model_retry_policies = retry_policies
+        self._repack()
+
+        # Clean up the PipelineSession Context
+        model.sagemaker_session.context = None
+
+        if self._regis_model_args:
+            register_model_step = _RegisterModelStep(
+                name="{}-RegisModel".format(name),
+                step_args=self._regis_model_args,
+                display_name=display_name,
+                retry_policies=self._register_model_retry_policies,
+                description=description,
+            )
+            if not self._regis_model_runtime_repack:
+                register_model_step.add_depends_on(depends_on)
```

Comment:
so, if I give both register model step arg, and create model arg, they will run in parallel during execution after all steps in this `depends_on` list are done? wouldn't create model step should be run prior to register model?

Reply:
also, create model step and register model step should depend on the repack model step right? if there is a runtime repack.

Reply:
> so, if I give both register model step arg, and create model arg, they will run in parallel during execution after all steps in this depends_on list are done?
Yes

> also, create model step and register model step should depend on the repack model step right? if there is a runtime repack.
If there's a runtime repack, the register/create model step should reference the repack step properties, which will introduce the dependent relationship.